File: v1/src/main/java/com/api/v1/services/AllRentsDeletionServiceImpl.java
package com.api.v1.services;

import com.api.v1.domain.RentRepository;
import com.api.v1.exceptions.EmptyRentEntityException;
import com.api.v1.mappers.RentResponseMapper;
import com.api.v1.utils.RentFinderUtil;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;

@Service
class AllRentsDeletionServiceImpl implements AllRentsDeletionService {

    @Autowired
    private RentRepository repository;

    @Autowired
    private RentFinderUtil rentFinderUtil;

    @Override
    public Mono<Void> deleteAll() {
        return repository
                .findAll()
                .hasElements()
                .flatMap(hasElements -> {
                    if (!hasElements) return Mono.error(EmptyRentEntityException::new);
                    return repository.deleteAll();
                });
    }

    @Override
    public Mono<Void> deleteByOrderNumber(String orderNumber) {
        return rentFinderUtil
                .find(orderNumber)
                .flatMap(rent -> repository.delete(rent));
    }

}

File: v2/src/main/kotlin/com/api/v2/rent/AllRentsDeletionServiceImpl.kt
package com.api.v2.rent

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.count
import kotlinx.coroutines.withContext
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
private class AllRentsDeletionServiceImpl: AllRentsDeletionService {

    @Autowired
    lateinit var rentFinderUtil: RentFinderUtil

    @Autowired
    lateinit var rentRepository: RentRepository

    override suspend fun deleteAll() {
        return withContext(Dispatchers.IO) {
            if (rentRepository.findAll().count() == 0) {
                throw EmptyRentEntityException()
            }
            rentRepository.deleteAll()
        }
    }

    override suspend fun deleteByOrderNumber(orderNumber: String) {
        return withContext(Dispatchers.IO) {
            val existingRent = rentFinderUtil.find(orderNumber)
            rentRepository.delete(existingRent)
        }
    }

}